### ISSUE TITLE
Expand ESLint with further rules

### ISSUE THREAD

**Issue by andrewtavis** [activist-org/activist#544]
### Terms

- [X] I have searched [open and closed feature requests](https://github.com/activist-org/activist/issues?q=is%3Aissue+label%3Afeature)
- [X] I agree to follow activist's [Code of Conduct](https://github.com/activist-org/activist/blob/main/.github/CODE_OF_CONDUCT.md)

### Description

This issue is to discuss and add new ESLint rules and also to refactor the codebase as needed to make it compliant after the changes :)

### Contribution

@Robert-Kovalcuk had some ideas of some rules that we should add. Let's discuss and we can check in with some others and go from there!

**Comment by andrewtavis** [comment]
CC @to-sta :)

**Comment by Robert-Kovalcuk** [comment]
Yes, I'll definitely have a look at that, out of the top of my mind I think, we should enforce multi-word components

**Comment by andrewtavis** [comment]
Thanks @Robert-Kovalcuk!

**Comment by to-sta** [comment]
@Robert-Kovalcuk good idea 💯.

**Comment by to-sta** [comment]
Checking for type `any` would be interesting. Maybe there is a rule for disallowing type `any`.

**Comment by andrewtavis** [comment]
CC @lucianmurmurache: this is the issue I was talking with you about. @Robert-Kovalcuk, @lucianmurmurache is adding things into the ESLint setup to expand a11y linting. Figured adding him into the discussion makes sense 😊

**Comment by andrewtavis** [comment]
Hey @Robert-Kovalcuk 👋 Checking in on this one as it definitely would be good to expand the rules again soon :) Let me know if you have time for a call soon!

**Comment by andrewtavis** [comment]
Note that in the above PR and commit I added in all essential rules for Vue as well as those from strongly recommended and recommended that made sense, namely:

```js
"vue/html-closing-bracket-newline": "error",
"vue/multiline-html-element-content-newline": "error",
"vue/singleline-html-element-content-newline": "error",
```

Closing this one 😊 Thanks for the discussion here and the help!
